index.js

```diff
@@ -23,24 +20,26 @@ export default (sbp('sbp/selectors/register', {
       eventQueues[name] = { events: [] }
     }
     const events = eventQueues[name].events
-    events.push({
+    const thisEvent = {
       sbpInvocation,
-      state: STATE_PENDING,
       promise: null
-    })
+    }
+    events.push(thisEvent)
     while (events.length > 0) {
       const event = events[0]
-      if (event.state === STATE_PENDING) {
-        event.state = STATE_INVOKING
-        event.promise = sbp(...event.sbpInvocation)
-        await event.promise
-        event.state = STATE_FINISHED
-      } else if (event.state === STATE_INVOKING) {
+      if (event === thisEvent) {
+        try {
+          const promise = sbp(...event.sbpInvocation)
+
+          event.promise = promise instanceof Promise ? new Promise((accept) => promise.finally(accept)) : Promise.resolve()
```

Comment:
Might as well resolve it with `Promise.resolve(promise)` right?

Reply:
Thought about that. I think this might be faster, although I'll run a benchmark.

Alternatives I considered:

1. `event.promise = Promise.resolve(promise).catch(() => {})`: wraps promise unnecessarily
2. Leaving that line as is, and doing `const promise = Promise.resolve(sbp(...))` above. Also wraps unnecessarily, but it might work and is cleaner (the wrapping happens anyhow on the return)

Reply:
No I meant just at the part at the end, after `:`.

You wanted to make it possible to catch exceptions right?

Reply:
The exceptions are delivered to the calling code. Using Promise.resolve after the `:` wouldn't do much in terms of exception handling (because in sync code `sbp` would have already failed and this line never executed). But, I simplified this part based on the other issue, so this line doesn't exist anymore.